projects/ngx-speculoos/src/lib/matchers.ts

```diff
@@ -1,226 +1,119 @@
-import { TestTextArea } from './test-textarea';
-import { TestInput } from './test-input';
-import { TestElement } from './test-element';
-import { TestSelect } from './test-select';
+import {TestTextArea} from './test-textarea';
+import {TestInput} from './test-input';
+import {TestElement} from './test-element';
+import {TestSelect} from './test-select';
 
-const speculoosMatchers: jasmine.CustomMatcherFactories = {
+export type MatcherFunction = (isNegative: boolean, el: any, expected?: string|number) => ({ pass: boolean; message: string });
```

Comment:
Do we agree that the type of `expected` is `string | number` (please add a space aroud the pipe symbol, BTW) because we only have matchers checking for an expected string or number, but that it could in fact be anything.
Wouldn't it be cleaner to define it as `any` from the start, then?

Reply:
I hesitated on this one. Right now, we only have `string | number | undefined`, so I choose to be strict with this definition. 

From my point of view, I like API which reduces the type instead of using `any`.